crates/subspace-proof-of-time/src/clock_maker.rs

```diff
@@ -0,0 +1,108 @@
+//! Clock maker related functionality.
+
+use crate::aes::AESWrapper;
+use crate::pot::{AesKey, AesSeed, PotConfig, ProofOfTime};
+use sc_client_api::BlockImportNotification;
+use sp_core::{Blake2Hasher, Hasher, H256};
+use sp_runtime::traits::{Block as BlockT, Header, NumberFor};
+
+pub const INITIAL_KEY: [u8; 16] = [
+    0x9a, 0x84, 0x94, 0x0f, 0xfe, 0xf5, 0xb0, 0xd7, 0x01, 0x99, 0xfc, 0x67, 0xf4, 0x6e, 0xa2, 0x7a,
+];
+
+pub struct ClockMakerImpl<Block> {
```

Comment:
I do not think this belongs to the library. How randomness is injected is kind of irrelevant to the proof of time itself. `ProofOfTime` should have something like `fn inject_randomness` and not care where it comes from, know nothing about nodes or farmers. This is just a low-level library meant to be used elsewhere.

Look at `subspace-proof-of-space`. It doesn't depend on Substrate, doesn't know anything about consensus. It just uses a hash function and a few new types that are fundamental part of the protocol and thus belong to `subspace-core-primitives` crate. Otherwise it is a completely generic implementation. This should be similar to that.

Reply:
Refactored